libraries/rodeos/rodeos.cpp

```diff
@@ -120,16 +120,21 @@ void rodeos_db_snapshot::end_block(const get_blocks_result_base& result, bool fo
    if (!writing_block || result.this_block->block_num != *writing_block)
       throw std::runtime_error("call start_block first");
 
-   bool near       = result.this_block->block_num + 4 >= result.last_irreversible.block_num;
+   // if we are trying to get caught up to head and are already near (within 4 blocks) of LIB only then call db->flush to disk
+   bool near       = false;
+   if (result.this_block->block_num < result.last_irreversible.block_num)
+      near = result.this_block->block_num + 4 >= result.last_irreversible.block_num;
```

Comment:
Once you are caught up with the chain, you never flush again? Seems like you will want a configurable flush threshold.

If I understand this `if`, you only flush for blocks: LIB-4, LIB-3, LIB-2, LIB-1. That seems like a really odd thing to do.

Reply:
1. Agree that having a configurable flush threshold will be useful. Will put that in. 

2. Agree with your comments on the odd looking `if` check. It was there from the prior implementation and I did not want to change that prior to demonstrating the fix. Once we get confirmation that the fix stands I will update this `if` condition as well.